crates/sail-plan/src/extension/function/size.rs

```diff
@@ -0,0 +1,108 @@
+/// [Credit]: <https://github.com/apache/datafusion/blob/b10b820acb6ad92b5d69810e3d4de0ef6f2d6a87/datafusion/functions-nested/src/cardinality.rs>
+use std::any::Any;
+use std::sync::Arc;
+
+use datafusion::arrow::array::{Array, ArrayRef, GenericListArray, OffsetSizeTrait, UInt64Array};
+use datafusion::arrow::datatypes::DataType;
+use datafusion_common::cast::{as_large_list_array, as_list_array, as_map_array};
+use datafusion_common::{exec_err, plan_err, Result};
+use datafusion_expr::{
+    ArrayFunctionSignature, ColumnarValue, ScalarUDFImpl, Signature, TypeSignature, Volatility,
+};
+
+use crate::extension::function::functions_nested_utils::{
+    compute_array_dims, make_scalar_function,
+};
+
+// expr_fn::cardinality doesn't fully match expected behavior.
+// Spark's cardinality function seems to be the same as the size function.
+// `cardinality`: https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.cardinality.html
+// `size`: https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.size.html
```

Comment:
I've added back a deleted comment here for better context. Let me know if it is still relevant.